ui/eslint.config.base.js

```diff
@@ -0,0 +1,36 @@
+// eslint plugin on vscode cannot resolve with just `require("@itwin/eslint-plugin")`
+const iTwinPlugin = require("../common/temp/node_modules/@itwin/eslint-plugin");
```

Comment:
importing like this so that vscode eslint works properly. @aruniverse lmk what you think. Thanks!

Reply:
seems hacky, theres prob a better way. plz investigate

Reply:
Here are some approaches I tried using:
1. Using the `@rushstack/eslint-patch/modern-module-resolution` library, which was also being used before but it might be unsupported now in newer ESLint version so getting error message both in VS Code ESLint extension and cli run: 
![image](https://github.com/iTwin/appui/assets/98850418/66fe71b5-220c-4726-96a2-7ff13564a2a6) Hence, this cannot be a solution
1. Using `eslint.options` in `.settings.json` like so: 
![image](https://github.com/iTwin/appui/assets/98850418/2c4ab8c8-cc17-4785-afc7-96851940b688)
This was being used in the core repo. However, this option seems to have been removed entirely and does not work at all. Output from ESLint VS Code extension: 
![image](https://github.com/iTwin/appui/assets/98850418/92adbb61-3099-43a1-b7b9-dee349c0fb96)
So this is not an option either.
1. We can have the base config use only custom rules and each config file for each package import itwin/eslint-plugin every time. The change would look like so: 
![image](https://github.com/iTwin/appui/assets/98850418/4ec92f11-4892-4435-a459-edc4942375ef)
Note that each `eslint.config.js` importing `eslint.config.base.js` would have to make those additions, essentially having multiple repetitions (violates DRY). This solution works for both cli runs and VS Code ESLint extension but it has the tradeoff of repetition like I mentioned above. 

At this point, any workaround I think will be somewhat hacky or need to have repetition like I mentioned. There might be a better solution in the future or I could have just missed something completely obvious, which is always a possibility.

Let me know if you wanna go with Option 3 or we should go with a different approach @aruniverse . Thanks!

Reply:
try moving this based config to the root of the repo
and then at the repo root package.json add the itwin eslint plugin pkg
and then try updating the require to as it should be, require(itwin/eslint-plugin), and see if that works

Reply:
> VS Code Extension still showing the error message: Cannot find module '@itwin/eslint-plugin'

did you restart the eslint server? use the command pallette to do so
theres no vscode eslint extension settings to point to the correct installation?

Reply:
Yup, I have been restarting the ESLint server using command pallette with every smallest change I make and verifying by adding an unused var on a tsx file. (Error shows up for unused var if ESLint working on VSCode). 

Yes, I have tried different variations with `eslint.nodePath` and `eslint. workingDirectories` settings and also the `resolvePluginsRelativeTo` as mentioned above with no luck so far. Most errors showing up when I tweak these settings are errors mentioning how the config file could not be found

Reply:
Another potential option would be to create a private package in the ui folder that exports the base config. Making it private would have it not be published and allow you to set eslint-plugin as a dependency, helping to fix the issue of having to point to ../common/temp/node_modules

Reply:
That might very well solve the issue at hand but creating an entirely new private package with a single dep and no code just to solve a vs code es lint issue seems pretty hacky to me and imo isn't the most simple and right solution here

Reply:
The biggest issue seems to be placement in reference to node_modules.The change to a flat config means that directly requiring configs and other items is now the way to do it. We would have to place it in a location that has access to a node_modules folder otherwise we would have to do the referencing to the common folder. 

From the above options i like 3 the most since you could place it anywhere and avoid the reference issues. it does require some repetition like you say but it seems like the best option due to the removal of the other settings that we used before

Reply:
I somewhat agree as well. Since each of these packages already have `@itwin/eslint-plugin` in their list of dependencies already.

I pushed this implementation on my recent commit, we can always revert this commit if we choose to not go with it